packages/service-worker/worker/test/happy_spec.ts

```diff
@@ -1738,117 +1738,161 @@ describe('Driver', () => {
+                .addFile('/index.html', '<script src="foo.hash.js"></script>')
+                .addFile('/foo.hash.js', 'console.log("FOO");')
+                .build());
+
+        const updatedServer = generateMockServerState(
+            new MockFileSystemBuilder()
+                .addFile('/index.html', '<script src="bar.hash.js"></script>')
+                .addFile('/bar.hash.js', 'console.log("BAR");')
+                .build());
+
+        // Create initial server state and initialize the SW.
+        scope = new SwTestHarnessBuilder().withServerState(originalServer).build();
+        driver = new Driver(scope, scope, new CacheDatabase(scope, scope));
 
-    it('enters degraded mode when unrecoverable state', async () => {
-      const originalFiles = new MockFileSystemBuilder()
-                                .addFile('/index.html', '<script src="foo.hash.js"></script>')
-                                .addFile('/foo.hash.js', 'console.log("FOO");')
-                                .build();
+        // Verify that all three clients are able to make the request.
+        expect(await makeRequest(scope, '/foo.hash.js', 'client1')).toBe('console.log("FOO");');
+        expect(await makeRequest(scope, '/foo.hash.js', 'client2')).toBe('console.log("FOO");');
+        expect(await makeRequest(scope, '/foo.hash.js', 'client3')).toBe('console.log("FOO");');
 
-      const updatedFiles = new MockFileSystemBuilder()
-                               .addFile('/index.html', '<script src="bar.hash.js"></script>')
-                               .addFile('/bar.hash.js', 'console.log("BAR");')
-                               .build();
+        await driver.initialized;
+        originalServer.clearRequests();
+
+        // Verify that the `foo.hash.js` file is cached.
+        expect(await makeRequest(scope, '/foo.hash.js')).toBe('console.log("FOO");');
+        originalServer.assertNoRequestFor('/foo.hash.js');
+
+        // Update the server to the updated version.
+        scope.updateServerState(updatedServer);
+        expect(await driver.checkForUpdate()).toEqual(true);
+
+        // Update the first two clients to the second server, keep `client3` as is.
+        const [client1, client2] =
+            await Promise.all([scope.clients.get('client1'), scope.clients.get('client2')]);
+
+        await driver.updateClient(client1);
+        await driver.updateClient(client2);
+
+        const latestServer = generateMockServerState(
+            new MockFileSystemBuilder()
+                .addFile('/index.html', '<script src="baz.hash.js"></script>')
+                .addFile('/baz.hash.js', 'console.log("BAZ");')
+                .build());
+
+        // Update the server to the latest version
+        scope.updateServerState(latestServer);
+        expect(await driver.checkForUpdate()).toEqual(true);
+
+        // Remove `foo.hash.js` from the cache to emulate the browser evicting files from the cache.
```

Comment:
As discussed offline, this is not ideal (removing all caches instead of just removing `foo.hash.js` from the cache), but is good enough for our purposes here. Please add a comment explaining that. For example:

```suggestion
        // Remove `foo.hash.js` from the cache to emulate the browser evicting files from the cache.
        // NOTE: For simplicity, we are clearing all caches instead of just removing `foo.hash.js`.
        //       This is not ideal, but OK for our purposes here.
```

(Same for similar lines in other tests in this `describe()` block.)